lib/widgets/home.dart

```diff
@@ -2,96 +2,528 @@ import 'package:flutter/material.dart';
-                child: const Text("#test here")), // scaffolding hack, see above
-            ],
+      appBar: AppBar(),
+      body: const LoadingPlaceholder(),
+      floatingActionButton: TextButton(
+        onPressed: () => Navigator.push(context,
+          MaterialWidgetRoute(page: const ChooseAccountPage())),
+        child: Text(zulipLocalizations.chooseAccountPageTitle)));
+  }
+}
+
+class _NavigationBarButton extends StatelessWidget {
+  const _NavigationBarButton({
+    required this.icon,
+    required this.selected,
+    required this.onPressed,
+  });
+
+  final IconData icon;
+  final bool selected;
+  final void Function() onPressed;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    final iconColor = WidgetStateColor.fromMap({
+      WidgetState.pressed:  designVariables.iconSelected,
+      ~WidgetState.pressed: selected ? designVariables.iconSelected
+                                     : designVariables.icon,
+    });
+
+    return AnimatedScaleOnTap(
+      scaleEnd: 0.875,
+      duration: const Duration(milliseconds: 100),
+      child: IconButton(
+        icon: Icon(icon, size: 24),
+        onPressed: onPressed,
+        style: IconButton.styleFrom(
+          // TODO(#417): Disable splash effects for all buttons globally.
+          splashFactory: NoSplash.splashFactory,
+          highlightColor: designVariables.iconSelected.withValues(alpha: 0.05),
+          shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(4))),
+        ).copyWith(foregroundColor: iconColor)));
+  }
+}
+
+void _showMenu(BuildContext context, {
+  required ValueNotifier<HomePageTab> tabNotifier,
+}) {
+  final menuItems = <Widget>[
+    // TODO(#252): Search
+    // const SizedBox(height: 8),
+    _InboxButton(tabNotifier: tabNotifier),
+    // TODO: Recent conversations
+    const _MentionsButton(),
+    const _StarredMessagesButton(),
+    const _CombinedFeedButton(),
+    // TODO: Drafts
+    _ChannelsButton(tabNotifier: tabNotifier),
+    _DirectMessagesButton(tabNotifier: tabNotifier),
+    // TODO(#1094): Users
+    const _MyProfileButton(),
+    const _SwitchAccountButton(),
+    // TODO(#198): Set my status
+    // const SizedBox(height: 8),
+    // TODO(#97): Settings
+    // TODO(#661): Notifications
+    // const SizedBox(height: 8),
+    // TODO(#1095): VersionInfo
+  ];
+
+  final designVariables = DesignVariables.of(context);
+  final accountId = PerAccountStoreWidget.accountIdOf(context);
+  showModalBottomSheet<void>(
+    context: context,
+    // Clip.hardEdge looks bad; Clip.antiAliasWithSaveLayer looks pixel-perfect
+    // on my iPhone 13 Pro but is marked as "much slower":
+    //   https://api.flutter.dev/flutter/dart-ui/Clip.html
+    clipBehavior: Clip.antiAlias,
+    useSafeArea: true,
+    isScrollControlled: true,
+    backgroundColor: designVariables.bgBotBar,
```

Comment:
Hmm yeah, that's a bug.

I don't see a good way to avoid it with the existing API of `showModalBottomSheet` and friends, though. (If we wanted the same custom background color for all our bottom sheets, or all our modal bottom sheets, we could do that with ThemeData.bottomSheetTheme. But this has a different background color from the message/topic/etc. action sheets.) So let's file that as a follow-up issue for post-launch.

Reply:
If we do not pass `backgroundColor` on the `showModalBottomSheet` call, we can set the background color on a `DecoratedBox` and have the `DesignVariables.of(context)`'s inline, or do something like:

```dart
  showModalBottomSheet<void>(
    // [...]
    builder: (BuildContext context) {
      final designVariables = DesignVariables.of(context);
      return PerAccountStoreWidget(
        accountId: accountId,
        child: DecoratedBox(
          decoration: BoxDecoration(color: designVariables.bgBotBar),
```

This does not set the color of the bottom sheet `Material`. So we also have to set `badckgroundColor` to something like `transparent` to avoid rendering artifacts:

![image](https://github.com/user-attachments/assets/992740eb-0555-4cbe-8d7a-4be62c90ce3b)

(with `backgroundColor: Colors.blue`)

Reply:
Yeah, and the `Material` also gets involved (and uses its color) in a variety of interactions, which we'd effectively be turning off. So that feels like a workaround approach that's likely to have subtle bugs of its own. I'm OK with just having that glitch if the theme happens to change just at the moment that you have the main menu open, and leaving the fix for it as a future issue.